packages/ogmios/src/ogmiosToCore/tx.ts

```diff
@@ -65,319 +53,305 @@ const mapPoolParameters = (poolParameters: Schema.PoolParameters): Cardano.PoolP
     owners: poolParameters.owners.map((ownerKeyHash) =>
       Cardano.createRewardAccount(Crypto.Ed25519KeyHashHex(ownerKeyHash), Cardano.addressNetworkId(rewardAccount))
     ),
+    pledge: poolParameters.pledge.lovelace,
     relays: poolParameters.relays.map(mapRelay),
     rewardAccount,
-    vrf: Cardano.VrfVkHex(poolParameters.vrf)
+    vrf: Cardano.VrfVkHex(poolParameters.vrfVerificationKeyHash)
   };
 };
 
 const mapCertificate = (certificate: Schema.Certificate): Cardano.Certificate => {
-  if ('stakeDelegation' in certificate) {
-    return {
-      __typename: Cardano.CertificateType.StakeDelegation,
-      poolId: Cardano.PoolId(certificate.stakeDelegation.delegatee),
-      stakeKeyHash: Crypto.Ed25519KeyHashHex(certificate.stakeDelegation.delegator)
-    };
-  }
-  if ('stakeKeyRegistration' in certificate) {
-    return {
-      __typename: Cardano.CertificateType.StakeKeyRegistration,
-      stakeKeyHash: Crypto.Ed25519KeyHashHex(certificate.stakeKeyRegistration)
-    };
+  switch (certificate.type) {
+    case 'stakeDelegation':
+      return {
+        __typename: Cardano.CertificateType.StakeDelegation,
+        // REVIEW: how can the stakePool be undefined?
```

Comment:
REWIVEW: Clarify this scenario

Reply:
Could open an issue/discussion in ogmios repo to report/ask

Reply:
@mirceahasegan

Reply:
I opened [Q&A topic in ogmios](https://github.com/CardanoSolutions/ogmios/discussions/339)